nmigen/lib/cdc.py

```diff
@@ -70,7 +72,27 @@ def elaborate(self, platform):
 
 
 class ResetSynchronizer:
+    """Synchronize the deassertion of a reset to a local clock.
+
+    Output `assertion` is asynchronous, so the local clock need not be free-running.
+
+    Parameters
+    ----------
+    arst : Signal(1), out
+        Asynchronous reset signal, to be synchronized.
+    domain : str
+        Name of domain to synchronize reset to.
+    n : int, >=1
+        Number of clock edges from input deassertion to output deassertion
```

Comment:
This is really the number of flops, isn't it? Since the important part is MTBF.

Reply:
You're right, MTBF is the important part here, delay is the job of CRG or some custom reset control logic. Will fix. Am away at the moment.